app/workers/commit_monitor_handlers/commit/gemfile_checker.rb

```diff
@@ -32,21 +36,65 @@ def tag
     "<gemfile_checker />"
   end
 
+  def label_name
+    LABEL_NAME
+  end
+
   def process_branch
     send("process_#{branch.pull_request? ? "pr" : "regular"}_branch")
   end
 
   def process_pr_branch
-    logger.info("#{self.class.name}##{__method__} Updating pull request #{branch.pr_number} with Gemfile comment.")
+    @pr = branch.pr_number
+    logger.info("#{self.class.name}##{__method__} Updating pull request #{pr} with Gemfile comment.")
 
     branch.repo.with_github_service do |github|
-      github.issues.comments.create(
-        :issue_id => branch.pr_number,
-        :body     => "#{tag}#{Settings.gemfile_checker.pr_contacts.join(" ")} Gemfile changes detected in commit #{branch.commit_uri_to(commit)}.  Please review."
-      )
+      @github = github
+      delete_pr_comments(pr_gemfile_comments)
+      add_pr_gemfile_comment
+      add_pr_label
     end
   end
 
+  def github_org_repo
+    [branch.repo.upstream_user, branch.repo.name]
+  end
+
+  def add_pr_label
+    issue_labels = github.issues.labels.list(*github_org_repo, "issue_id" => pr)
```

Comment:
The branch should be associated with a PR, so you can get the issue first, and then get the labels and not need the repo info I think...Plus the github handle comes from the branch object with *should* already have this info...

Something like

```ruby
issue = github.issues("issue_id" => pr)
issue.labels.list
```